controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -118,6 +119,29 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (controller.OperationRe
 		return controller.ContinueProcessing()
 	}
 
+	// The default value for ErrorOccured is 'True'.  We know that the state
+	// of the condition is still 'True' due to the earlier call to
+	// HaveBindingsFailed().  If this condition is still true after a
+	// reasonable time then we assume that the SEB is stuck in an unrecoverable
+	// state and clean it up.  Otherwise we requeue and wait until the timeout
+	// has passed
+	const timeoutMinutes float64 = 5
+	var lastTransitionTime time.Time
+	for _, condition := range a.snapshotEnvironmentBinding.Status.BindingConditions {
```

Comment:
While this loop works as intended, I think that using `meta.FindStatusCondition()` (like [here](https://github.com/redhat-appstudio/integration-service/blob/32b1591224e1708e6d934d9f855f9c8675dab9ca/gitops/binding.go#L69)) could help readability.